docs/build/voting-dapp-tutorial.md

```diff
@@ -0,0 +1,1245 @@
+
+:::info
+**Gas** is the cost necessary to perform a transaction on the network. Gas is paid to miners and its price varies based on supply and demand. It's a critical piece of the puzzle, but at the same time it brings up a UX problem. Every user needs to be aware of what gas is as well as how much gas they need to pay for their transaction. This causes significant friction and a less than ideal experience.
+
+To help mitigate this problem Kadena brings an innovation to the game. Hello **gas stations**! Gas stations are a way for dApps to subsidize gas costs for their users. This means that your user doesn't need to know what gas is or how much the gas price is, which translates into a smooth experience when interacting with your dApp.
+:::
+
+### Voting
+
+Let's start to implement our voting contract.
+
+A typical developer workflow looks like this:
+
+1. Write logic in `.pact` files
+2. Write tests in `.repl` files
+3. Execute your tests in the repl
+4. Deploy to local pact server
+5. Deploy to testnet
+
+Here we will be focusing on steps 1-3.
+
+Assuming you are still in your project directory, create the `election.pact` and `election.repl` files:
+
+```clojure
+touch pact/election.pact pact/election.repl
+```
+
+`election.pact` is the contract source code while in `election.repl` we'll write tests.
+
+We will also need the `coin` contract which requires the `fungible-v2` interface for our tests. You can get the latest `coin-v3` [here](https://github.com/kadena-io/chainweb-node/blob/master/pact/coin-contract/v3/coin-v3.pact) and `fungible-v2` [here](https://github.com/kadena-io/chainweb-node/blob/master/pact/coin-contract/v2/fungible-v2.pact). Make sure to add those files in your project.
+
+Now copy the following code in the `election.pact` file:
+
+```clojure
+(define-keyset 'election-admin-keyset)
+
+(module election GOVERNANCE
+  "Election demo module"
+
+  (defcap GOVERNANCE ()
+    "Only admin can update this module"
+    (enforce-keyset 'election-admin-keyset))
+
+  (defun vote (key)
+    "Submit a new vote"
+    (format "Voted {}!" [key]))
+)
+
+(vote "A")
+```
+
+This snippet defines a module "election" that holds a function "vote" that takes a parameter and displays a formatted string that includes the parameter.
+
+:::note
+Module names and keyset definitions are required to be unique. We will mention this again when we get to deploy our contract to testnet.
+:::
+
+Before we move forward, let's quickly test our code. Copy the code below in the `election.repl` file:
+
+```clojure
+;; begin a transaction
+(begin-tx)
+;; set transaction signature key to my-key
+(env-keys ["my-key"])
+;; Add the election-admin-keyset to environment data
+(env-data {
+  'election-admin-keyset: { "keys": ["admin-key"], "pred": "keys-all" }  
+})
+;; load election.pact into the REPL
+(load "election.pact")
+;; commit the transaction
+(commit-tx)
+```
+
```

Comment:
Add another ::info or ::note that explains that each pact transaction is submitted with raw json data that functions can read from, and where we can store our keysets

Reply:
Not sure what you mean here: "where we can store our keysets"